cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -109,13 +121,134 @@ public Void call() throws Exception {
     }
   }
 
+  /**
+   * Clean log files which does not have corresponding meta data
+   * @param tillTime time till the meta data will be deleted.
+   * @param namespacedLogBaseDirMap namespace to directory map
+   * @param parentDirs parent directories for deleted files
+   * @throws Exception
+   */
+  private void cleanFilesWithoutMeta(final long tillTime, final Map<String, NamespaceId> namespacedLogBaseDirMap,
+                                     final SetMultimap<String, Location> parentDirs) throws Exception {
+    LOG.info("Starting deletion of log files older than {} without metadata", tillTime);
+    List<NamespaceMeta> namespaceMetaList = namespaceQueryAdmin.list();
+    for (NamespaceMeta namespaceMeta : namespaceMetaList) {
+      final NamespaceId namespaceId = namespaceMeta.getNamespaceId();
+      final Location namespacedLogBaseDir = LoggingContextHelper.getNamespacedBaseDirLocation(namespacedLocationFactory,
+                                                                                              logBaseDir, namespaceId,
+                                                                                              impersonator);
+      if (namespacedLogBaseDir.exists()) {
+        final Processor<LocationStatus, Set<Location>> processor = getLocationProcessor(namespaceId, tillTime);
+        impersonator.doAs(namespaceId, new Callable<Void>() {
+          @Override
+          public Void call() throws Exception {
+            Locations.processLocations(namespacedLogBaseDir, true, processor);
+            return null;
+          }
+        });
+        Set<Location> result = processor.getResult();
```

Comment:
Would be good to name `result` as `nonMetaFiles` or something similar for readability.